### PR TITLE
Group communication preferences

### PR DESCRIPTION
**Pull request by ErikSchierboom**
- Remove weird id from inputs
- Split communication preferences into groups

image

### CODE REVIEW

File: app/javascript/components/settings/CommunicationPreferencesForm.tsx [hunk 1]
```diff
@@ -60,24 +66,37 @@ export const CommunicationPreferencesForm = ({
...
+          />
+          <div className="row">
+            <div className="c-checkbox">
+              <GraphicalIcon icon="checkmark" />
+            </div>
+            {p.label}
+          </div>
+        </label>
+      ))}
+      <h2>Product</h2>
```
Comment (ErikSchierboom) [inline app/javascript/components/settings/CommunicationPreferencesForm.tsx:84]: Let me know what you think of the grouping (and their names).

File: app/javascript/components/settings/ShowOnSupportersPageButton.tsx [hunk 2]
```diff
@@ -39,12 +39,7 @@ export const ShowOnSupportersPageButton = ({
 
   return (
     <label className="c-checkbox-wrapper ml-auto">
-      <input
-        type="checkbox"
-        id="communication_preferences_email_on_mentor_started_discussion_notification"
```
Comment (ErikSchierboom) [inline app/javascript/components/settings/ShowOnSupportersPageButton.tsx:44]: This ID was quite weird to see here.

Comment (iHiD) [reply app/javascript/components/settings/ShowOnSupportersPageButton.tsx:44]: It's been copied/pasted from Rails which autogenerates them.

**Review by iHiD — APPROVED** [review]
Nice work :)

**Pull request merged**